src/CodeGen_Hexagon.cpp

```diff
@@ -569,7 +578,24 @@ Value *CodeGen_Hexagon::slice_vector(Value *vec, int start, int size) {
                                         {vec});
             }
         }
-        // TODO: Could maybe use valign to implement this?
+        // We know vec is a double vector.
+        int bytes_off = start * (element_bits / 8);
+        int reverse_bytes = (native_vector_bits() / 8) - bytes_off;
+        Intrinsic::ID IntrinsID = IPICK(Intrinsic::hexagon_V6_valignbi);
+        if (reverse_bytes <= 7) {
+            IntrinsID = IPICK(Intrinsic::hexagon_V6_vlalignbi);
+            bytes_off = reverse_bytes;
+        }
+        Value *hi, *lo;
+        if (call) {
```

Comment:
call being non-null doesn't guarantee that it's a call to vcombine (note the condition above is (call && call->getCalledFunction() == combine_fn)